zokrates_core/src/flatten/mod.rs

```diff
@@ -357,12 +357,35 @@ impl Flattener {
 
                 FlatExpression::Identifier(name_1_y)
             },
+            BooleanExpression::AndAnd(box lhs, box rhs) => {
+                let x = self.flatten_boolean_expression(
+                    functions_flattened,
+                    arguments_flattened,
+                    statements_flattened,
+                    lhs
+                );
+                let y = self.flatten_boolean_expression(
+                    functions_flattened,
+                    arguments_flattened,
+                    statements_flattened,
+                    rhs
+                );
+                let name_x_and_y = self.use_sym();
+
+                // flatten_boolean_expression always returns a linear term
+                assert!(x.is_linear() && y.is_linear());
+                statements_flattened.push(FlatStatement::Definition(
```

Comment:
This new definition is not needed in theory as x*y is guaranteed to be quadratic, but I see how it makes sure the return is an identifier and we can compose boolean expressions. Maybe we could have a comment to that effect?